detection_rules/misc.py

```diff
@@ -148,33 +148,69 @@ def _convert_type(_val):
 
 def get_kibana_rules_map(branch='master'):
     """Get list of available rules from the Kibana repo and return a list of URLs."""
-    r = requests.get('https://api.github.com/repos/elastic/kibana/branches?per_page=1000')
-    branch_names = [b['name'] for b in r.json()]
-    if branch not in branch_names:
-        raise ValueError('branch "{}" does not exist in kibana'.format(branch))
+    # ensure branch exists
+    r = requests.get(f'https://api.github.com/repos/elastic/kibana/branches/{branch}')
+    if not r.ok:
+        raise ValueError(f'Kibana branch: {branch} does not exist')
```

Comment:
could be a different error altogether. This would be true for a 404. but raise_for_status() could be better

Reply:
I can tighten down to 404 👍

Reply:
went with your suggestion